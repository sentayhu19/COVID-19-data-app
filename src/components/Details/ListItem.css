:root {
  --darkblue: #24498f;
  --lightblue: #4872c2;
  --white: rgb(254, 254, 254);
}

.country {
  border: 1px solid black;
  height: 173px;
}

.country:nth-child(odd) {
  float: left;
  width: 50%;
}

.covid-icn {
  width: 45px;
  height: 42px;
}

.arrow-right {
  position: relative;
  top: 0;
  width: 25px;
  color: white;
  text-align: right;
  margin-left: 70%;
}

.country-name {
  color: var(--white);
  width: 145px;
}

.total-confirmed {
  color: var(--white);
}

.country-list:nth-child(4n-1),
.country:nth-child(4n-2) {
  background-color: var(--darkblue);
}

.country:nth-child(4n+4) {
  background-color: var(--lightblue);
}

.country-list:nth-child(4n),
.country:nth-child(4n+1) {
  background-color: var(--lightblue);
}

.country:nth-child(3),
.country:nth-child(8n+7),
.country:nth-child(8n+11) {
  background-color: var(--darkblue);
}

.summary {
  display: flex;
  flex-direction: column;
  float: right;
}
