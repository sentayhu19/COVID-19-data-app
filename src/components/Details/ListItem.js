import React from 'react';
import { NavLink } from 'react-router-dom';
import './ListItem.css';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight } from '@fortawesome/free-solid-svg-icons';
import covid from '../../images/covid.png';

const ListItem = ({ countries, TotalConfirmed }) => {
  const passData = countries;
  return (
    <div className="country">
      <div className="summary">
        <NavLink id={countries} to="/Details" state={{ country: passData }}>
          <FontAwesomeIcon icon={faCircleArrowRight} className="arrow-right" />
          <img src={covid} className="covid-icn" alt="COVID-19" />
          <p className="country-name">{countries}</p>
          <p className="total-confirmed">{(TotalConfirmed).toLocaleString('en-US')}</p>
        </NavLink>
      </div>
    </div>
  );
};
ListItem.propTypes = {
  countries: PropTypes.string.isRequired,
  TotalConfirmed: PropTypes.number.isRequired,
};
export default ListItem;
