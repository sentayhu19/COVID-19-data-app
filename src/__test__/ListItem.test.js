import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import ListItem from '../components/Details/ListItem';

describe('Test for List of each Country', () => {
  test('Country List Item Renders correctly: ', () => {
    const totalConfirmed = '10000';
    const ListCountry = renderer.create(
      <MemoryRouter>
        <ListItem countries="Ethiopia" TotalConfirmed={totalConfirmed} />
      </MemoryRouter>,
    ).toJSON();
    expect(ListCountry).toMatchSnapshot();
  });
});
