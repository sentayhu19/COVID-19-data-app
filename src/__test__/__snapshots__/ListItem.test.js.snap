// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for List of each Country Country List Item Renders correctly:  1`] = `
<div
  className="country"
>
  <div
    className="summary"
  >
    <a
      className=""
      href="/Details"
      id="Ethiopia"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-circle-arrow-right arrow-right"
        data-icon="circle-arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 0C114.6 0 0 114.6 0 256c0 141.4 114.6 256 256 256s256-114.6 256-256C512 114.6 397.4 0 256 0zM406.6 278.6l-103.1 103.1c-12.5 12.5-32.75 12.5-45.25 0s-12.5-32.75 0-45.25L306.8 288H128C110.3 288 96 273.7 96 256s14.31-32 32-32h178.8l-49.38-49.38c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l103.1 103.1C414.6 241.3 416 251.1 416 256C416 260.9 414.6 270.7 406.6 278.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <img
        alt="COVID-19"
        className="covid-icn"
        src="covid.png"
      />
      <p
        className="country-name"
      >
        Ethiopia
      </p>
      <p
        className="total-confirmed"
      >
        10000
      </p>
    </a>
  </div>
</div>
`;
